l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -0,0 +1,306 @@
+    ) external view override returns (bool) {
+        address stateTransition = getStateTransition(_chainId);
+        return IZkSyncStateTransition(stateTransition).proveL2LogInclusion(_batchNumber, _index, _log, _proof);
+    }
+
+    /// @notice forwards function call to Mailbox based on ChainId
+    function proveL1ToL2TransactionStatus(
+        uint256 _chainId,
+        bytes32 _l2TxHash,
+        uint256 _l2BatchNumber,
+        uint256 _l2MessageIndex,
+        uint16 _l2TxNumberInBatch,
+        bytes32[] calldata _merkleProof,
+        TxStatus _status
+    ) external view override returns (bool) {
+        address stateTransition = getStateTransition(_chainId);
+        return
+            IZkSyncStateTransition(stateTransition).proveL1ToL2TransactionStatus(
+                _l2TxHash,
+                _l2BatchNumber,
+                _l2MessageIndex,
+                _l2TxNumberInBatch,
+                _merkleProof,
+                _status
+            );
+    }
+
+    /// @notice forwards function call to Mailbox based on ChainId
+    function l2TransactionBaseCost(
+        uint256 _chainId,
+        uint256 _gasPrice,
+        uint256 _l2GasLimit,
+        uint256 _l2GasPerPubdataByteLimit
+    ) external view returns (uint256) {
+        address stateTransition = getStateTransition(_chainId);
+        return
+            IZkSyncStateTransition(stateTransition).l2TransactionBaseCost(
+                _gasPrice,
+                _l2GasLimit,
+                _l2GasPerPubdataByteLimit
+            );
+    }
+
+    /// @notice the mailbox is called directly after the sharedBridge received the deposit
+    /// this assumes that either ether is the base token or
+    /// the msg.sender has approved mintValue allowance for the sharedBridge.
+    /// This means this is not ideal for contract calls, as the contract would have to handle token allowance of the base Token
+    function requestL2TransactionDirect(
+        L2TransactionRequestDirect calldata _request
+    ) external payable override nonReentrant returns (bytes32 canonicalTxHash) {
+        {
+            address token = baseToken[_request.chainId];
+            if (token == ETH_TOKEN_ADDRESS) {
+                require(msg.value == _request.mintValue, "Bridgehub: msg.value mismatch");
+                // kl todo it would be nice here to be able to deposit weth instead of eth
+                sharedBridge.bridgehubDepositBaseToken{value: _request.mintValue}(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+            } else {
+                require(msg.value == 0, "Bridgehub: non-eth bridge with msg.value");
+                // note we have to pass token, as a bridge might have multiple tokens.
+                sharedBridge.bridgehubDepositBaseToken(_request.chainId, msg.sender, token, _request.mintValue);
+            }
+        }
```

Comment:
You can remove code deduplication here by moving `sharedBridge.bridgehubDepositBaseToken` from the branching
```suggestion
            if (token == ETH_TOKEN_ADDRESS) {
                require(msg.value == _request.mintValue, "Bridgehub: msg.value mismatch");
            } else {
                require(msg.value == 0, "Bridgehub: non-eth bridge with msg.value");
            }
            
            sharedBridge.bridgehubDepositBaseToken{value: msg.value}(
                    _request.chainId,
                    msg.sender,
                    token,
                    _request.mintValue
                );
        }
```